Detectors/ZDC/calib/src/InterCalibSpec.cxx

```diff
@@ -0,0 +1,191 @@
+// Copyright 2019-2020 CERN and copyright holders of ALICE O2.
+// See https://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+// All rights not expressly granted are reserved.
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// @file   InterCalibSpec.cxx
+/// @brief  ZDC reconstruction
+/// @author pietro.cortese@cern.ch
+
+#include <iostream>
+#include <vector>
+#include <string>
+#include "CCDB/BasicCCDBManager.h"
+#include "CCDB/CCDBTimeStampUtils.h"
+#include "Framework/Logger.h"
+#include "Framework/ControlService.h"
+#include "Framework/ConfigParamRegistry.h"
+#include "Framework/CCDBParamSpec.h"
+#include "DetectorsCommonDataFormats/DetID.h"
+#include "DataFormatsZDC/BCData.h"
+#include "DataFormatsZDC/ChannelData.h"
+#include "DataFormatsZDC/OrbitData.h"
+#include "DataFormatsZDC/RecEvent.h"
+#include "ZDCBase/ModuleConfig.h"
+#include "CommonUtils/NameConf.h"
+#include "CCDB/BasicCCDBManager.h"
+#include "CCDB/CCDBTimeStampUtils.h"
+#include "ZDCReconstruction/RecoConfigZDC.h"
+#include "ZDCReconstruction/ZDCEnergyParam.h"
+#include "ZDCReconstruction/ZDCTowerParam.h"
+#include "ZDCCalib/InterCalibSpec.h"
+
+using namespace o2::framework;
+
+namespace o2
+{
+namespace zdc
+{
+
+InterCalibSpec::InterCalibSpec()
+{
+  mTimer.Stop();
+  mTimer.Reset();
+}
+
+InterCalibSpec::InterCalibSpec(const int verbosity) : mVerbosity(verbosity)
+{
+  mTimer.Stop();
+  mTimer.Reset();
+}
+
+void InterCalibSpec::init(o2::framework::InitContext& ic)
+{
+  //     int minEnt = std::max(300, ic.options().get<int>("min-entries"));
+  //     int nb = std::max(500, ic.options().get<int>("nbins"));
+  //     int slotL = ic.options().get<int>("tf-per-slot");
+  //     int delay = ic.options().get<int>("max-delay");
+  mVerbosity = ic.options().get<int>("verbosity-level");
+  mInterCalib.setVerbosity(mVerbosity);
+}
+
+void InterCalibSpec::updateTimeDependentParams(ProcessingContext& pc)
+{
+  // we call these methods just to trigger finaliseCCDB callback
+  pc.inputs().get<o2::zdc::ZDCEnergyParam*>("energycalib");
```

Comment:
It crashes because you don't requiest `energycalib` input in the `getInterCalibSpec()`.

Reply:
Thanks for spotting this!